osf/metadata/utils.py

```diff
@@ -23,44 +23,81 @@
 }
 
 
+def datacite_format_name_identifiers(user):
+    data = {
+        'nameIdentifiers': [
+            {
+                'nameIdentifier': f'{settings.DOMAIN}{user._id}/',
+                'nameIdentifierScheme': 'URL',
+            }
+        ]
+    }
+
+    if user.external_identity.get('ORCID'):
+        verified = list(user.external_identity['ORCID'].values())[0] == 'VERIFIED'
+        if verified:
+            data['nameIdentifiers'].append({
+                'nameIdentifier': list(user.external_identity['ORCID'].keys())[0],
+                'nameIdentifierScheme': 'ORCID',
+                'schemeURI': 'http://orcid.org/'
+            })
+
+    if user.social_links.get('orcid'):
+        data['nameIdentifiers'].append({
+            'nameIdentifier': user.social_links.get('orcid'),
+            'nameIdentifierScheme': 'ORCID',
+            'schemeURI': 'http://orcid.org/'
+        })
+
+    return data
+
+
+def datacite_format_affilations(user):
+    data = {'affiliation': []}
+    for affiliated_institution in user.affiliated_institutions.all():
+        data['affiliation'].append({
+            'name': affiliated_institution.name,
+        })
+        for domain in affiliated_institution.domains:
```

Comment:
short term: remove this, as these domains seem OSF-specific. Ticket to be made for resolving how to store actual institutional domains for metadata